src/associations/belongs-to-many.ts

```diff
@@ -0,0 +1,1316 @@
+      this.throughModel.rawAttributes[this.otherKey].primaryKey = true;
+    } else if (this.through.unique !== false) {
+      let uniqueKey;
+      if (typeof this.through.unique === 'string' && this.through.unique !== '') {
+        uniqueKey = this.through.unique;
+      } else {
+        const keys = [this.foreignKey, this.otherKey].sort();
+        uniqueKey = [this.through.model.tableName, ...keys, 'unique'].join('_');
+      }
+
+      this.throughModel.rawAttributes[this.foreignKey].unique = uniqueKey;
+      this.throughModel.rawAttributes[this.otherKey].unique = uniqueKey;
+    }
+
+    this.throughModel.refreshAttributes();
+  }
+
+  static associate<
+    S extends Model,
+    T extends Model,
+    ThroughModel extends Model,
+    SourceKey extends AttributeNames<S>,
+    TargetKey extends AttributeNames<T>,
+  >(
+    secret: symbol,
+    source: ModelStatic<S>,
+    target: ModelStatic<T>,
+    options: BelongsToManyOptions<SourceKey, TargetKey, ThroughModel>,
+    pair?: BelongsToMany<T, S, ThroughModel, TargetKey, SourceKey>,
+    parent?: Association<any>,
+  ): BelongsToMany<S, T, ThroughModel, SourceKey, TargetKey> {
+    return defineAssociation<
+      BelongsToMany<S, T, ThroughModel, SourceKey, TargetKey>,
+      BelongsToManyOptions<SourceKey, TargetKey, ThroughModel>,
+      NormalizedBelongsToManyOptions<SourceKey, TargetKey, ThroughModel>
+    >(BelongsToMany, source, target, options, parent, normalizeOptions, newOptions => {
+      // self-associations must always set their 'as' parameter
+      if (isSameInitialModel(source, target)
+        // use 'options' because this will always be set in 'newOptions'
+        && (!options.as || !options.inverse?.as || options.as === options.inverse.as)) {
+        throw new AssociationError('Both options "as" and "inverse.as" must be defined for belongsToMany self-associations, and their value must be different.');
+      }
+
+      return new BelongsToMany(secret, source, target, newOptions, pair, parent);
+    });
+  }
+
+  #mixin(modelPrototype: Model) {
+
+    mixinMethods(
+      this,
+      modelPrototype,
+      ['get', 'count', 'hasSingle', 'hasAll', 'set', 'add', 'addMultiple', 'remove', 'removeMultiple', 'create'],
+      {
+        hasSingle: 'has',
+        hasAll: 'has',
+        addMultiple: 'add',
+        removeMultiple: 'remove',
+      },
+    );
+  }
+
+  /**
+   * Get everything currently associated with this, using an optional where clause.
+   *
+   * See {@link Model} for a full explanation of options
+   *
+   * @param instance instance
+   * @param options find options
+   */
+  async get(instance: SourceModel, options?: BelongsToManyGetAssociationsMixinOptions<TargetModel>): Promise<TargetModel[]> {
+    const through = this.through;
+
+    const findOptions: FindOptions<Attributes<TargetModel>> = {
+      ...options,
+      // TODO: current WhereOptions typings do not allow having 'WhereOptions' inside another 'WhereOptions'
```

Comment:
Just for completeness;  I made a comment about this in one of the other files as well. That same comment applies here